### PR TITLE
[Backport main] 2.3.0 Release notes

### PR DESCRIPTION
**Pull request by dreamer-89**
Manual backport of https://github.com/opensearch-project/job-scheduler/pull/237 to `main`

### CODE REVIEW

**Review by joshpalis — APPROVED** [review]

**Comment by dreamer-89** [comment]
@saratvemulapalli @joshpalis : Waiting for one more review on this PR for merge.

**Review by saratvemulapalli — APPROVED** [review]

**Pull request merged**
